Add support for Debian 9, drop debian 8

bastelfreak opened the pull request:
<!--
Thank you for contributing to this project!

- This project has a Contributor Code of Conduct: https://voxpupuli.org/coc/
- Please check that here is no existing issue or PR that addresses your problem.
- Our vulnerabilities reporting process is at https://voxpupuli.org/security/

-->

ekohl commented on .travis.yml:
```diff
@@ -33,14 +33,19 @@ matrix:
+    sudo: required
+  - rvm: 2.4.2
+    bundler_args: --without development
+    env: PUPPET_VERSION="~> 5.0" CHECK=acceptance BEAKER_set="docker/debian-9" bundle exec rake acceptance
```
```beaker-puppet_install_helper``` uses a different syntax. How this even worked in the past is a mystery to me but if you read https://github.com/puppetlabs/beaker-puppet_install_helper#run_puppet_install_helper then it suggests you need explicit version numbers.

bastelfreak replied:
to clarify: This never worked and was a copy and paste issue

ekohl commented:
I honestly think setting Puppet versions is pretty broken with the acceptance tests. I'd much rather set the repo collection to pc1 or puppet5. For not just leave it out and use the default so we at least test something on debian 5?

bastelfreak commented:
Current state: In general everything is working. Also this PR fixes https://github.com/voxpupuli/puppet-gluster/issues/140. Debian 8 seems to be a bit strange. After the package installation the daemon is started, but not via unit file (which is also missing in 3.8, but that is EOL anyway)? So puppet tries to start the daemon, bind to the tcp socket fails because it is already running -> everything is broken. Personally I would drop Debian 8 support. It is legacy, Debian 9 is working. Objections?

bastelfreak commented:
Also the unit files are really really bad. They still rely on the legacy sysv scripts. The mentionted issues result in the usage of official glusterfs repos on Debian 8. Maybe it works if we use the debian-own packages for glusterfs, on debian 8 only. But implementing that would be really really painful and adds a lot of additional logic to the modul. Also the shipped glusterfs version in Debian 8 is, of course, EOL.

ekohl commented:
👍 to dropping Debian Jessie.

ekohl commented:
This need a rebase now. Could you also add the nodesets to ```.sync.yml```?

bastelfreak commented:
I also dropped debian 8 from the metadata.json. We already dropped official support for it and purged it from the travis acceptance test matrix. We just forgot the entry in the metadata.json
